crypto/bn/bn_lib.c

```diff
@@ -649,15 +649,27 @@ int BN_clear_bit(BIGNUM *a, int n)
 int BN_is_bit_set(const BIGNUM *a, int n)
 {
     int i, j;
+    BN_ULONG s;
 
     bn_check_top(a);
     if (n < 0)
         return 0;
     i = n / BN_BITS2;
     j = n % BN_BITS2;
-    if (a->top <= i)
-        return 0;
-    return (int)(((a->d[i]) >> j) & ((BN_ULONG)1));
+
+    if (a->flags & BN_FLG_CONSTTIME) {
+        int it;
+
+        if (i >= a->dmax)
+            return 0;
+        it = constant_time_lt(i, a->top);
```

Comment:
I think this should use a->dmax, not a->top. As far as I know everything between top and dmax should be set to 0.

Reply:
And so this line should just get removed ...

Reply:
Sadly, no, stuff between `top` and `dmax` is not set to zero. For instance, this function:
https://github.com/openssl/openssl/blob/master/crypto/bn/bn_lib.c#L854-L859

(This is why I'm not a huge fan of playing `top` vs `dmax` games.)

Reply:
It has to use top, as @davidben wrote, there is no guarantee of zeroes between top and gmax.  The code has to behave as if there are zeroes there but still read what's really there...